lib/hubspot/company.rb

```diff
@@ -185,8 +197,7 @@ def [](property)
     # @param params [Hash] hash of properties to update
     # @return [Hubspot::Company] self
     def update!(params)
-      query = {"properties" => Hubspot::Utils.hash_to_properties(params.stringify_keys!, key_name: "name")}
-      Hubspot::Connection.put_json(UPDATE_COMPANY_PATH, params: { company_id: vid }, body: query)
+      self.class.update!(vid, params)
       @properties.merge!(params)
```

Comment:
I don't think this is a blocker for this PR since we didn't have this behavior already but we're only updating the properties with the changes from `params` which means we're not updating dynamic properties such as `timestamp` and `versions`. I think the best approach is to copy the properties from the returned object to the current object which is what `ActiveRecord` [reload()](https://github.com/rails/rails/blob/536a190ab3690810a3b342b897f2585c4971229d/activerecord/lib/active_record/persistence.rb#L608) does.

Thoughts?

Reply:
Interesting catch 🎣  I agree that this isn't a blocker for this PR.

Based on your description, I'm imagining something like:
```
response = self.class.update!(vid, params)

# Re-initializes the object based on a hash of values from the API call
# and updates the state of the object
refresh_from(response)
```

Reply:
As noted by @fonji below, the Hubspot just returns a 204 so there isn't really anything in the response to refresh from is there?

Reply:
The Hubspot APIs are inconsistent with their return results from updates. Some only return a 204, some return a partial record with the changed fields, and others return the entire record.

I’ve mostly worked this functionality out on the master branch but I’ve not yet pushed it since I didn’t finish the tests yet.